lib/Form/FormsRepository.php

```diff
@@ -27,6 +27,20 @@ public function findAllNotDeleted(): array {
       ->getResult();
   }
 
+  public function getNamesOfFormsForSegments(): array {
+    $allNonDeletedForms = $this->findAllNotDeleted();
```

Comment:
There is but I believe it is discouraged. The tables are mapped to the models and retrieving partial values means we get entities that are missing data. As the table is unlikely to get big, I would not optimize it.
Again, I am also not familiar with doctrine, [read about it on this article](https://www.doctrine-project.org/projects/doctrine-orm/en/2.9/reference/partial-objects.html#partial-objects), so happy to change it if I am wrong.

Reply:
Thanks for pointing me to this article. I was not aware of it. To me, it makes sense to avoid using partial objects as the article suggests unless we really need to which is not the case here.

Going forward, for situations similar to this one, I will consider using `getArrayResult()` instead of `getResult()`. It seems like a way to retrieve just what we need without running into the partial objects problem.